library/alloc/src/rc.rs

```diff
@@ -3536,7 +3536,7 @@ impl<T> Default for Weak<T> {
     }
 }
 
-// NOTE: We checked_add here to deal with mem::forget safely. In particular
+// NOTE: We wrapping_add here to deal with mem::forget safely. In particular
```

Comment:
This is confusing in a different way: the `wrapping_add` does not offer the protection anymore, the `strong == 0` conditional does.

Reply:
at the very least, it should mention clearly what actually implements the guard that handles the `mem::forget`, if it's going to mention any code.

Reply:
I see what you mean. The fact that it's `wrapping_add` rather than `checked_add` previously is merely a detail for the sake of code generation, and that's addressed in a more local comment. In that case, I think this comment is more about the higher level issue of a potential soundness hole and isn't enhanced by mentioning the particular implementation details. How about this?
```
// NOTE: If you mem::forget Rcs (or Weaks), drop is skipped and the ref-count
// is not decremented, meaning the ref-count can overflow, and then you can
// free the allocation while outstanding Rcs (or Weaks) exist, which would be
// unsound. We abort because this is such a degenerate scenario that we don't
// care about what happens -- no real program should ever experience this.
```

Reply:
Yeah, something like that seems good to me.